### DISCUSSION TITLE
Can't download video from Udemy in Chrome.

### DISCUSSION THREAD

**Discussion by oblogin** [aclap-dev/video-downloadhelper#207] [Help and Q&A] [answered]
### Your question

Hi, I can't download video from Udemy in Chrome after last update. The extension finds the video, but after clicking the download button, an error shows up. I can download videos from other sites. I don't use adblock, folders has write permission. Is there any way to fix the download?

### Addon & CoApp version and details

Error: Exit code: 1
    at rs.receive (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:13157)
    at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42624
--- 

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: ru
license: accepted
key (only 16 first characters): 9d01fd3a62114c14
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\oblog",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

Video DownloadHelper
Версия 8.2.1.1 (stable)
Локаль броузера: ru
Технический билд
Скомпилировано 2024-02-19
Опции компиляции: browser=chrome, linuxlic=false, noyt=true, target=google

Платформа Win x86-64
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36

Найдено приложение-компаньон: VdhCoApp 2.0.19
Исполняемый файл компаньона: C:\Program Files\DownloadHelper CoApp\vdhcoapp.exe

**Comment by GrampaWildWilly** [comment]
Start on the page you can get by clicking the Discussions link at the top of this page.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/1475db54-ded2-4909-b0b1-0c5ca6fb9709)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/a2f35cf8-6d8f-4010-b0df-b657468f25da)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f2f22ff9-b637-4e5f-80d0-d2a644775fdf)

**Reply by GrampaWildWilly** [reply]
@jcfrog Hi Jérôme.  My images above could be used for a tutorial to add to the Wiki.  The title could be:

How to search for your answer before you post

You could just steal these images or you could be inspired by them to create images of your own.

**Reply by GrampaWildWilly** [reply]
Here's a second chapter of the same tutorial.  It's the same tutorial, not a different one.  This is another way to do the same search.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0c338d4a-1bcd-46e4-9312-22a20e725627)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2c4c2fe0-c025-4ab6-b013-caf385ce1c03)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/2fc62a6f-cba7-4501-a926-3f6d4025d7b8)
![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/19ed5a7c-47d1-4626-859d-25dfddc783c3)
![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/38424446-d094-4f45-aabe-d4be03943456)

**Comment by oblogin** [comment]
I realize you have a whole bunch of posts every day. But again, just a week ago all the videos uploaded. I downloaded 160 videos. Even now Firefox can download videos from Udemy and there is no error. A week ago I haven't problems with the quality or size of the videos. If this error can't be fixed, you can say that Udemy videos can't be downloaded in Chrome like Youtube videos can't be downloaded. And yes, I read the posts related to Udemy and this bug and you said you will fix it in the 8.2.0.24 update.

https://github.com/aclap-dev/video-downloadhelper/discussions/91
https://groups.google.com/g/video-downloadhelper-q-and-a/c/n1noFczTFAM

**Reply by GrampaWildWilly** [reply]
Goodness.  That's over a year ago.  Plus, it seems I have not made it clear that I do not work for VDH.  I'm just another user, like you.  So I did not say I would fix anything.  I can't.  It is entirely possible that Udemy changed their policy.  I'm not a member there so I haven't been monitoring things on there.  All I can say is that one of our fellow users discovered what you can read in the posts on THIS forum.  That says some Udemy content will download & some won't.  Like I said, I'm not a member over there so I can't investigate this any further.

**Comment by paulrouget** [comment]
I need a link to the failing udemy video.

**Reply by azizloj** [reply]
can't download any video from udemy right now :(

**Reply by GrampaWildWilly** [reply]
You need to check if the content is protected by DRM.  This is easy to do in Firefox.  Firefox places an icon in the URL bar at the top of window.  Hover your mouse over the icons to the left of the URL.  Each one gives a hover up tooltip.  Read the tooltips.  If one of them mentions DRM, then the content is protected by DRM & cannot be downloaded.  I don't know how to do it in Chrome & Edge.  I have no experience with those browsers.

I made a general comment about this issue in #984.

**Reply by oblogin** [reply]
> can't download any video from udemy right now :(
So far, I've found an alternative solution.
https://4saved.com/

**Reply by oblogin** [reply]
> You need to check if the content is protected by DRM...

There is no DRM protection. Checked in Firefox.

**Reply by azizloj** [reply]
> > can't download any video from udemy right now :(
> > So far, I've found an alternative solution.
> > https://4saved.com/

no longer available for chrome :(

**Comment by paulrouget** [comment]
The Udemy problem is known, we are working on solution. Hope to get that resolved next week.

**Comment by paulrouget** [comment, marked as answer]
Can someone tells me if this coapp fixes the issue: https://github.com/aclap-dev/vdhcoapp/releases/download/v2.0.20/vdhcoapp-windows-x86_64-installer.exe

**Reply by oblogin** [reply]
Tested it on 5 videos, it works.

**Reply by azizloj** [reply]
> Can someone tells me if this coapp fixes the issue: https://github.com/aclap-dev/vdhcoapp/releases/download/v2.0.20/vdhcoapp-windows-x86_64-installer.exe

Unfortunately, it didn't work. Yes, it downloaded the video, but it's blank—no image or sound.

**Reply by Mattmjs** [reply]
Sounds like your video has DRM protection which is unfortunate.

**Reply by azizloj** [reply]
Thats Udemy :-)

**Reply by devinschumacher** [reply]
> Can someone tells me if this coapp fixes the issue: https://github.com/aclap-dev/vdhcoapp/releases/download/v2.0.20/vdhcoapp-windows-x86_64-installer.exe

Does not solve it on Mac ARM @paulrouget 

chrome version 125.0.6422.61



<img width="556" alt="image" src="https://github.com/user-attachments/assets/e4427c9d-5f46-4146-9c8f-f52e9b983093">


<img width="547" alt="image" src="https://github.com/user-attachments/assets/70db0acf-ad5d-4f50-b388-483ca00c8b35">
